docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
+  - [Proposed Features](#Proposed-Features)
+- [Product scope](#Product-scope)
+- [User Stories](#User-Stories)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
+
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The design of Developer Guide is referenced from the [AB-3 Developer Guide](http://se-education.org/addressbook-level3/DeveloperGuide.html).
+
+
+
+## Design & Implementation
+### Architecture
+#### Main components of the architecture
+![image info](./UmlDiagrams/Structure.png)  
+
+**Main** contains the main class called `CsProjPlanner`. It is responsible for:
+* At app launch: invokes methods to initialize the app, and gets prepared for receiving user input.
+* At shut down: invokes methods to store generated data.
+
+**UI** contains two classes `Constants` and `Response`. It is responsible for:
+* Store messages which will be displayed frequently.
+* Store magic numbers.
+* Handles the responses to give to user.  
+
+**Logic Component**  
+![image info](./UmlDiagrams/Logic.png)
+How the `Logic` component works:
+* When `CommandHandler` is called upon to execute a command, it creates `XYZCommandParser` object(e.g.,`AddProjectCommandParser`) which parses the user command and create a `XYZCommand` object (e.g., `AddProjectCommand`).
+* The created `XYZCommand` object is returned as a `Command` object.
+* All `XYZCommandParser` classes(e.g., `AddProjectCommandParser`, `AddTodoCommandParser`)inherit from the `Parser` interface.
+
+
+**Model Component**
+![image info](./UmlDiagrams/Model.png)
+
+* The `Model` component stores all project data i.e., all `Project` objects in a list i.e.`ProjectList`.  
+* Each project contains `Deadline` and `Todo` objects.  
+* Does not depend on any of the other components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
+
+
+**Storage** contains class `Storage`. It is responsible for:
+* Save project data in txt format, and read them back into corresponding objects.
+* Dependent on some classes in the `Model` component.
+
+
+### Implemented Features
+
+Following a restucture to the project to follow a more Object Oriented structure, as well as to allow for conformity and easy editing of code, many of our commands will follow a similar structure of the following:
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user
+
+The basic flow of the structure:
+
+![image info](./UmlDiagrams/CommandParser.png)
```

Comment:
For the sequence diagram for command parser, shouldn't the activation box for :AddProjectCommandParser end once the reply message ends?